src/pages/ConnectWallet/ConnectWallet.tsx

```diff
@@ -14,79 +15,79 @@ import OrbsStatic from 'assets/orbs-static.png'
 import { FoxIcon } from 'components/Icons/FoxIcon'
 import { Page } from 'components/Layout/Page'
 import { RawText, Text } from 'components/Text'
+import { KeyManager, SUPPORTED_WALLETS } from 'context/WalletProvider/config'
 import { ActionTypes, InitialState, WalletActions } from 'context/WalletProvider/WalletProvider'
 import { useQuery } from 'hooks/useQuery/useQuery'
 import { colors } from 'theme/colors'
 
-import { KeyManager, SUPPORTED_WALLETS } from '../../context/WalletProvider/config'
-
 type WalletProps = {
   state: InitialState
   dispatch: Dispatch<ActionTypes>
 }
 
+async function connectCypressWallet(
+  keyring: Keyring,
+  dispatch: Dispatch<ActionTypes>,
+  walletId: string,
+  walletPassword: string
+) {
+  const adapter = SUPPORTED_WALLETS[KeyManager.Native].adapter.useKeyring(keyring)
+  if (adapter) {
+    const wallet = await adapter.pairDevice(walletId)
+    await wallet.initialize()
```

Comment:
Add a comment above this line to capture the note you made in the PR:

```ts
// This will trigger the Password modal due to the MNEMONIC_REQUIRED native event sent by wallet.initialize(). 
// We can just ignore it as the wallet will be set programmatically.
```

Reply:
No `wallet.initialize()` anymore in current implementation.